src/memory.rs

```diff
@@ -470,11 +480,14 @@ impl<'a, 'tcx> Memory<'a, 'tcx> {
         }
     }
 
-    pub fn get_fn(&self, id: AllocId) -> EvalResult<'tcx, ty::Instance<'tcx>> {
-        debug!("reading fn ptr: {}", id);
-        match self.functions.get(&id) {
+    pub fn get_fn(&self, ptr: Pointer) -> EvalResult<'tcx, ty::Instance<'tcx>> {
+        if ptr.offset != 0 {
```

Comment:
I would say calling `offset` on them is UB -- and miri implements this, as the bounds-check will always fail.
Calling `wrapping_offset` OTOH must not be UB, as that's a safe function.